Tools/Scripts/libraries/webkitscmpy/webkitscmpy/commit_classifier.py

```diff
@@ -126,7 +136,9 @@ def classify(self, commit, repository=None):
             matches_header = klass.headers and header and any([f(header) for f in klass.headers])
             matches_trailers = klass.trailers and trailers and any([any([f(trailer) for f in klass.trailers]) for trailer in trailers])
             matches_content = klass.contents and contents and any([f(contents) for f in klass.contents])
-            if (matching_header or matching_trailer or matching_content) and not matches_header and not matches_trailers and not matches_content:
+            if (matching_header or matching_trailer) and not matches_header and not matches_trailers:
```

Comment:
This feels wrong…

Per [De Morgan's laws](https://en.wikipedia.org/wiki/De_Morgan%27s_laws), this is: `(matching_header or matching_trailer) and not (matches_header or matches_trailers)`, which is a contradiction.

Reply:
Actually I'm not sure why it's a contradiction. `(matching_header or matching_trailer)` checks whether the class actually defines a header or trailer to be matched with. Then, if we get no match for either one of them, we want to skip this class?

Might be silly, but I want to point out `matching_header != matches_header` just in case.

Reply:
> Might be silly, but I want to point out `matching_header != matches_header` just in case.

This is not silly, this is "you have a dyslexic reviewer". 🙃